src/libraries/System.Text.Json/src/System/Text/Json/JsonTokenType.cs

```diff
@@ -77,5 +77,14 @@ public enum JsonTokenType : byte
         ///   Indicates that the token type is the JSON literal <c>null</c>.
         /// </summary>
         Null,
+
+        /// <summary>
+        ///   Indicates that the token type is a segment of a JSON string.
+        /// </summary>
+        /// <remarks>
+        ///   This does not represent a token defined in the JSON specification, but rather provides a means to track
+        ///   that a string is being written in segments. This value will never be seen during deserialization.
+        /// </remarks>
+        StringSegment,
```

Comment:
This is not part of the approved public API. Instead of declaring a new member here, you can declare a `private const JsonTokenType StringSegmentSentinel = (JsonTokenType)-1;` inside `Utf8JsonWriter` and use that. The class' public API does not return `JsonTokenType`s so there is a low risk of accidentally returning these sentinel values to user code.

Reply:
I like this idea!

Reply:
Note that this enum is used by both the reader and writer types, so unless we add support for both this intermediate state should be internal to Utf8JsonWriter